include/socgraph.php

```diff
@@ -190,181 +187,166 @@ function poco_load_worker($cid, $uid, $zcid, $url) {
 	);
 
 }
+/**
+ * @brief Sanitize the given gcontact data
+ *
+ * @param array $gcontact array with gcontact data
+ *
+ * Generation:
+ *  0: No definition
+ *  1: Profiles on this server
+ *  2: Contacts of profiles on this server
+ *  3: Contacts of contacts of profiles on this server
+ *  4: ...
+ *
+ */
+function sanitized_gcontact(&$gcontact) {
 
-function poco_check($profile_url, $name, $network, $profile_photo, $about, $location, $gender, $keywords, $connect_url, $updated, $generation, $cid = 0, $uid = 0, $zcid = 0) {
-
-	// Generation:
-	//  0: No definition
-	//  1: Profiles on this server
-	//  2: Contacts of profiles on this server
-	//  3: Contacts of contacts of profiles on this server
-	//  4: ...
-
-	$gcid = "";
-
-	if ($profile_url == "")
-		return $gcid;
+	if ($gcontact['url'] == "") {
+		return false;
+	}
 
-	$urlparts = parse_url($profile_url);
-	if (!isset($urlparts["scheme"]))
-		return $gcid;
+	$urlparts = parse_url($gcontact['url']);
+	if (!isset($urlparts["scheme"])) {
+		return false;
+	}
 
 	if (in_array($urlparts["host"], array("www.facebook.com", "facebook.com", "twitter.com",
```

Comment:
It is hardcoded because we are having a handful of networks we are handling and it doesn't make any sense to change this in some configuration. It is like here: https://github.com/friendica/friendica/blob/61a01141d7e897c30b5931b66675cbb09d6368b8/include/contact_selectors.php#L75-L94

Here it doesn't make sense to have it configurable as well.